account-kit/smart-contracts/src/ma-v2/actions/deferralActions.test.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  erc7677Middleware,
+  LocalAccountSigner,
+  type SmartAccountSigner,
+} from "@aa-sdk/core";
+import {
+  custom,
+  parseEther,
+  publicActions,
+  testActions,
+  type TestActions,
+} from "viem";
+import { installValidationActions } from "@account-kit/smart-contracts/experimental";
+import {
+  // createModularAccountV2Client,
+  type SignerEntity,
+} from "@account-kit/smart-contracts";
+import { local070Instance } from "~test/instances.js";
+import { setBalance } from "viem/actions";
+import { accounts } from "~test/constants.js";
+import { alchemyGasAndPaymasterAndDataMiddleware } from "@account-kit/infra";
+import { deferralActions } from "./deferralActions.js";
+import { PermissionBuilder, PermissionType } from "../permissionBuilder.js";
+import { createModularAccountV2Client } from "../client/client.js";
+
+// Note: These tests maintain a shared state to not break the local-running rundler by desyncing the chain.
+describe("MA v2 deferral actions tests", async () => {
+  const instance = local070Instance;
+
+  let client: ReturnType<typeof instance.getClient> &
+    ReturnType<typeof publicActions> &
+    TestActions;
+
+  beforeAll(async () => {
+    client = instance
+      .getClient()
+      .extend(publicActions)
+      .extend(testActions({ mode: "anvil" }));
+  });
+
+  const signer: SmartAccountSigner = new LocalAccountSigner(
+    accounts.fundedAccountOwner
+  );
+
+  const target = "0x000000000000000000000000000000000000dEaD";
+  const sendAmount = parseEther("1");
+
+  it("tests the full deferred actions flow", async () => {
+    const provider = (await givenConnectedProvider({ signer }))
+      .extend(deferralActions)
+      .extend(installValidationActions);
+
+    await setBalance(instance.getClient(), {
+      address: provider.getAddress(),
+      value: parseEther("2"),
+    });
+
+    const sessionKeyEntityId = 1;
+    const sessionKey: SmartAccountSigner = new LocalAccountSigner(
+      accounts.unfundedAccountOwner
+    );
+
+    // these can be default values or from call arguments
+    const { entityId, nonce } = await provider.getEntityIdAndNonce({
+      entityId: 0,
+      nonceKey: 0n,
+      isGlobalValidation: true,
+    });
+
+    // TODO: remove nonce override here
+    const { typedData } = await new PermissionBuilder(provider)
+      .configure({
+        key: {
+          publicKey: await sessionKey.getAddress(),
+          type: "secp256k1",
+        },
+        entityId: sessionKeyEntityId,
```

Comment:
Should this be `entityId`?

Nit: I think we should start using more descriptive names, i.e. `nextEntityId` or `sessionKeyEntityId` or `ownerEntityId` just in general.